TypeSystem/Serialization/test.cc

```diff
@@ -323,17 +461,47 @@ TEST(Serialization, JSON) {
     EXPECT_EQ("LLandP", parsed.m.at("spock"));
     EXPECT_EQ("MTFBWY", parsed.m.at("jedi"));
   }
+  // Serializing an `std::unordered_map<>` with simple key type, which becomes a JSON object.
+  {
+    WithTrivialUnorderedMap with_unordered_map;
+    EXPECT_EQ("{\"m\":{}}", JSON(with_unordered_map));
+    with_unordered_map.m["foo"] = "fizz";
+    with_unordered_map.m["bar"] = "buzz";
+    const std::set<std::string> valid_jsons{"{\"m\":{\"bar\":\"buzz\",\"foo\":\"fizz\"}}",
```

Comment:
This is excellent, but comparing two `std::ordereded_map<>`-s via their own `operator==()` could work too.